daemon/CMakeLists.txt

```
@@ -27,13 +29,21 @@ add_executable(bpftime_daemon
     user/main.cpp
 )
 
-add_dependencies(libbpftime_daemon
-    bpftime_daemon_ebpf_skel
-    libbpf
-    spdlog::spdlog
-    runtime
-)
+if(${BPFTIME_BUILD_WITH_LIBBPF})
+    add_dependencies(libbpftime_daemon
+        bpftime_daemon_ebpf_skel
+        libbpf
+        spdlog::spdlog
+        runtime
+    )
+else()
+    add_dependencies(libbpftime_daemon
+        spdlog::spdlog
+        runtime
+    )
+endif()
```

Can we do...
```suggestion
f(${BPFTIME_BUILD_WITH_LIBBPF})
    add_dependencies(libbpftime_daemon
        bpftime_daemon_ebpf_skel
        libbpf
    )
endif()
add_dependencies(libbpftime_daemon
  spdlog::spdlog
  runtime
  )
```
? Not sure how `add_dependencies` work together on cmake

---

didn't understand this change.